x/evm/types/msg.go

```diff
@@ -242,18 +254,50 @@ func (msg MsgEthereumTx) GetSignBytes() []byte {
 func (msg MsgEthereumTx) RLPSignBytes(chainID *big.Int) ethcmn.Hash {
 	return rlpHash([]interface{}{
 		msg.Data.AccountNonce,
-		msg.Data.Price,
+		msg.Data.Price.BigInt(),
 		msg.Data.GasLimit,
-		msg.Data.Recipient,
-		msg.Data.Amount,
+		msg.To(),
+		msg.Data.Amount.BigInt(),
 		msg.Data.Payload,
 		chainID, uint(0), uint(0),
 	})
 }
 
 // EncodeRLP implements the rlp.Encoder interface.
 func (msg *MsgEthereumTx) EncodeRLP(w io.Writer) error {
-	return rlp.Encode(w, &msg.Data)
+	var hash ethcmn.Hash
+	if len(msg.Data.Hash) > 0 {
+		hash = ethcmn.HexToHash(msg.Data.Hash)
+	}
+
+	data := struct {
+		AccountNonce uint64
+		Price        *big.Int        `json:"gasPrice"`
+		GasLimit     uint64          `json:"gas"`
+		Recipient    *ethcmn.Address `json:"to" rlp:"nil"` // nil means contract creation
```

Comment:
can we just use `*ethcmn.Address` as the recipient type in the msg?

Reply:
no, because of the proto types we need to use string, and you can't take a string pointer on protobuf, so we need to wrap it on this `Recipient` struct